### PR TITLE
Wrap CoNvex Gap-cost alignMents for Long Reads (ngmlr) version 0.2.7

### PR DESCRIPTION
**Pull request by davebx**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection

### CODE REVIEW

File: tools/ngmlr/ngmlr.xml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        <param argument="--query" type="data" format="fasta,fastq,fasta.gz,fastq.gz" label="Reads to align" />
+        <param name="bam_fix" argument="--bam-fix" type="boolean" truevalue="--bam-fix" falsevalue="" checked="true" label="Report reads with &gt; 64k CIGAR operations as unmapped" help="Required to be compatible with the BAM format" />
+        <conditional name="sam_header">
+            <param name="header_data" type="select" label="SAM Read Group">
+                <option value="no">Do not set</option>
+                <option value="yes">Set values</option>
+            </param>
+            <when value="yes">
+                <param name="rg_id" argument="--rg-id" type="text" label="Value for the read group field in the SAM/BAM header" />
+                <param name="rg_sm" argument="--rg-sm" type="text" label="RG: Sample" />
```
Comment (bgruening) [inline tools/ngmlr/ngmlr.xml:58]: should we add validators here?

File: tools/ngmlr/ngmlr.xml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                <param name="rg_fo" argument="--rg-fo" type="text" label="RG: Flow order" />
+                <param name="rg_ks" argument="--rg-ks" type="text" label="RG: Key sequence" />
+            </when>
+            <when value="no" />
+        </conditional>
+        <param argument="--presets" type="select" label="Presets to use for alignment">
+            <option value="pacbio">PacBio</option>
+            <option value="ont">Nanopore</option>
+        </param>
+        <param name="min_identity" argument="--min-identity" type="float" min="0" max="1" value="0.65" label="Alignments with an identity lower than this threshold will be discarded" />
```
Comment (bgruening) [inline tools/ngmlr/ngmlr.xml:76]: I think you can remove all `name=` here ...

File: tools/ngmlr/ngmlr.xml (hunk 1 above)
Comment (davebx) [reply tools/ngmlr/ngmlr.xml:58]: Not a bad idea, do you think alphanumeric and spaces would be generous enough?

Comment (bgruening) [reply tools/ngmlr/ngmlr.xml]: Not sure. I had hope we have this already somewhere, but only could find: https://github.com/galaxyproject/tools-iuc/blob/91a4a5aacf1e8bac5ddccb8504e46dae13b594aa/macros/read_group_macros.xml#L229 and https://samtools.github.io/hts-specs/SAMtags.pdf

Maybe allow all printable and remove quotes from the set?

Comment (davebx) [reply tools/ngmlr/ngmlr.xml]: That sounds reasonable. I'll update this later today.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Merci!

**Comment by mschatz** [comment]
awesome!

**Pull request merged**